api/src/main/java/org/apache/iceberg/actions/MigrateTable.java

```diff
@@ -50,6 +50,16 @@ default MigrateTable dropBackup() {
     throw new UnsupportedOperationException("Dropping a backup is not supported");
   }
 
+  /**
+   * Sets a table name for the backup of the original table.
+   *
+   * @param tableName the table name for backup
+   * @return this for method chaining
+   */
+  default MigrateTable withBackupTableName(String tableName) {
```

Comment:
I think we should drop `with` prefix from the name given that no other existing methods have them.